companion.py

```diff
@@ -545,176 +694,356 @@ def auth_callback(self) -> None:
+            capi_data: CAPIData
+            if query.endpoint == self._CAPI_PATH_STATION:
+                try:
+                    capi_data = capi_station_queries()
+
+                except Exception as e:
+                    self.capi_response_queue.put(
+                        (
+                            CAPIFailedRequest(
+                                message=e.args,
+                                exception=e,
+                                query_time=query.query_time,
+                                play_sound=query.play_sound,
+                                auto_update=query.auto_update
+                            ),
+                        )
+                    )
 
-        A /profile query is performed to check that we are docked (or on foot)
-        and the station name and marketid match the prior Docked event.
-        If they do match, and the services list says they're present, also
-        retrieve CAPI market and/or shipyard/outfitting data and merge into
-        the /profile data.
+                else:
+                    self.capi_response_queue.put(
+                        EDMCCAPIResponse(
+                            capi_data=capi_data,
+                            query_time=query.query_time,
+                            play_sound=query.play_sound,
+                            auto_update=query.auto_update
+                        )
+                    )
+
+            else:
+                try:
+                    capi_data = capi_single_query(self.FRONTIER_CAPI_PATH_PROFILE)
+
+                except Exception as e:
+                    self.capi_response_queue.put(
+                        (
+                            CAPIFailedRequest(
+                                message=e.args,
+                                exception=e,
+                                query_time=query.query_time,
+                                play_sound=query.play_sound,
+                                auto_update=query.auto_update
+                            ),
+                        )
+                    )
 
-        :return: Possibly augmented CAPI data.
+                else:
+                    self.capi_response_queue.put(
+                        EDMCCAPIResponse(
+                            capi_data=capi_data,
+                            query_time=query.query_time,
+                            play_sound=query.play_sound,
+                            auto_update=query.auto_update
+                        )
+                    )
+
+            # If the query came from EDMC.(py|exe) there's no tk to send an
+            # event too, so assume it will be polling there response queue.
+            if query.tk_response_event is not None:
+                self.tk_master.event_generate('<<CAPIResponse>>')
+
+        logger.info('CAPI worker thread DONE')
+
+    def capi_query_close_worker(self) -> None:
+        """Ask the CAPI query thread to finish."""
+        self.capi_query_queue.put(None)
```

Comment:
do we maybe want a better sentinel here?

Reply:
Well, we could define a special '__EDMC_WORKER_SHUTDOWN' 'endpoint' and pass that in, and that would at least remove a need for 'Optional' on the queue member types declaration.

Reply:
That works for me